src/checkbox/base-checkbox.tsx

```diff
@@ -25,11 +25,7 @@ export interface BaseCheckboxProps extends BaseComponentProps, FormFieldControlP
   disabled?: boolean;
 
   /**
-   * Specifies the ID of the native form element. You can use it to relate
-   * a label element's `for` attribute to this control.
-   *
-   * It defaults to an automatically generated ID that
-   * is provided by its parent form field component.
+   * Specifies the ID of the native form element. It uses an automatically generated ID by default.
```

Comment:
Why the "You can use it to relate a label element's `for` attribute to this control" part was removed?

Reply:
I think we don't need highlight it(it feels like recommendation), because unlike Input, Checkbox and Toggle usually have their own labels to control the form.

Reply:
Sure, but as the controlId property exists I think it makes sense to describe its purpose fully. I'd imagine it is possible to provide an external label e.g. if you're building some selectable card or something alike.

If we want to discourage that use of the controlId for these elements I think it is better to deprecate this and provide an explanation.

Reply:
can you include @stirache in this too, as it impacts public-facing documentation

Reply:
The controlId can be used to access the input, not only to relate a label. So I don't discourage using it. But I am not keen to state this usage specifically, because it doesn't feel like a very common use case as they have their own label already. To build selectable cards, users should use card instead.

Reply:
@gethinwebster I will include Rachel when we have consensus what to include in the description.

Reply:
I think the controlId is specifically meant for control/label association. Why would you need to assign an id on the native input otherwise? For testing we recommend using our test-utils that can be combined with date-testid.

Reply:
I dont have the data what else users may need the controlId. But for example we have controlId in Select, which is not meant (as I understand) for control/label association.

Reply:
I think in select it works for the input/label association as well: https://github.com/cloudscape-design/components/blob/main/src/select/internal.tsx#L91.

```
// This line takes controlId from restProps or the form-field otherwise.
const formFieldContext = useFormFieldContext(restProps);
const selfControlId = useUniqueId('trigger');
const controlId = formFieldContext.controlId ?? selfControlId;
```

Reply:
As I understand, the select takes the id from formfield, but not meant for input/label association? Also it is not stated in the description https://github.com/cloudscape-design/components/blob/main/src/select/interfaces.ts#L91

Reply:
In that case I would rather challenge the select.controlId description to include that info.